dom.d.ts

```diff
@@ -0,0 +1,44 @@
+type Nil = null | undefined;
+type Primitive = string | number | bigint | boolean;
+
+type Prop = Nil | Primitive;
+type Props = Nil | Record<string, Prop>;
+
+type Child = Nil | Primitive | Node | Child[];
+
+export function E<N extends HTMLInputElement>(name: 'input', props?: Props, ...children: Child[]): N;
+export function E<N extends HTMLElement>(name: string, props?: Props, ...children: Child[]): N;
+
+export function S(name: string, props?: Props, ...children: Child[]): SVGElement;
+
+export function F(...children: Child[]): Node;
+
+export function reset<N extends Node>(node: N | Nil, props?: Props, ...children: Child[]): N | Nil;
```

Comment:
Currently this requires `N` and doesn't accept `Nil`. (I've been thinking about allowing that, not sure.)